src/analysis/complex/isometry.lean

```diff
@@ -0,0 +1,183 @@
+  ∥f z - 1∥ = ∥z - 1∥ :=
+by rw [←linear_isometry.norm_map f (z - 1), linear_isometry.map_sub, h]
+
+lemma linear_isometry.im_apply_eq_im {f : ℂ →ₗᵢ[ℝ] ℂ} (h : f 1 = 1) (z : ℂ) :
+  z + conj z = f z + conj (f z) :=
+begin
+  have := linear_isometry.abs_apply_sub_one_eq_abs_sub_one h z,
+  apply_fun λ x, x ^ 2 at this,
+  simp only [norm_eq_abs, ←norm_sq_eq_abs] at this,
+  rw [←of_real_inj, ←mul_conj, ←mul_conj] at this,
+  rw [conj.map_sub, conj.map_sub] at this,
+  simp only [sub_mul, mul_sub, one_mul, mul_one] at this,
+  rw [mul_conj, norm_sq_eq_abs, ←norm_eq_abs, linear_isometry.norm_map] at this,
+  rw [mul_conj, norm_sq_eq_abs, ←norm_eq_abs] at this,
+  simp only [sub_sub, sub_right_inj, mul_one, of_real_pow, ring_hom.map_one, norm_eq_abs] at this,
+  simp only [add_sub, sub_left_inj] at this,
+  rw [add_comm, ←this, add_comm],
+end
+
+lemma linear_isometry.re_apply_eq_re {f : ℂ →ₗᵢ[ℝ] ℂ} (h : f 1 = 1) (z : ℂ) : (f z).re = z.re :=
+begin
+  apply linear_isometry.re_apply_eq_re_of_add_conj_eq,
+  intro z,
+  apply linear_isometry.im_apply_eq_im h,
+end
+
+lemma linear_isometry_complex_aux (f : ℂ →ₗᵢ[ℝ] ℂ) (h : f 1 = 1) :
+  (∀ z, f z = z) ∨ (∀ z, f z = conj z) :=
+begin
+  have h0 : f I = I ∨ f I = -I,
+  { have : |f I| = 1,
+    { rw [←norm_eq_abs, linear_isometry.norm_map, norm_eq_abs, abs_I] },
+    simp only [ext_iff, ←and_or_distrib_left, neg_re, I_re, neg_im, neg_zero],
+    split,
+    { rw ←I_re,
+      rw linear_isometry.re_apply_eq_re h },
+    { apply linear_isometry.im_apply_eq_im_or_neg_of_re_apply_eq_re,
+      { intro z, rw [←norm_eq_abs, ←norm_eq_abs, linear_isometry.norm_map] },
+      { intro z, rw linear_isometry.re_apply_eq_re h, } } },
+  refine or.imp (λ h1, _) (λ h1 z, _) h0,
+  { suffices : f.to_linear_map = linear_isometry.id.to_linear_map,
+    { simp [this, ←linear_isometry.coe_to_linear_map, linear_map.id_apply] },
+    apply is_basis.ext is_basis_one_I,
+    intro i,
+    fin_cases i,
+    { simp [h] },
+    { simp only [matrix.head_cons, linear_isometry.coe_to_linear_map,
+        linear_map.id_coe, id.def, matrix.cons_val_one],
+      exact h1, } },
+  { suffices : f.to_linear_map = conj_li.to_linear_map,
+    { rw [←linear_isometry.coe_to_linear_map, this],
+      simp only [linear_isometry.coe_to_linear_map],
+      refl },
+    apply is_basis.ext is_basis_one_I,
+    intro i,
+    fin_cases i,
+    { simp only [h, linear_isometry.coe_to_linear_map, matrix.cons_val_zero],
+      change 1 = conj 1,
+      ext; simp },
+    { simp only [matrix.head_cons, linear_isometry.coe_to_linear_map,
+        linear_map.id_coe, id.def, matrix.cons_val_one],
+      change f I = conj I,
+      rw conj_I,
+      exact h1, } },
+end
+
+lemma linear_isometry_complex (f : ℂ →ₗᵢ[ℝ] ℂ) :
+  ∃ a : ℂ, |a| = 1 ∧ ((∀ z, f z = a * z) ∨ (∀ z, f z = a * conj z)) :=
+begin
+  let a := f 1,
+  use a,
+  split,
+    { change ∥a∥ = 1,
+      simp only [a],
+      rw linear_isometry.norm_map,
+      simp,
+    },
```

Comment:
```suggestion
  { change ∥a∥ = 1,
    simp only [a],
    rw linear_isometry.norm_map,
    simp },
```
Braces should be aligned with the previous indentation, and the last brace should not be alone on its line. Same thing everywhere in this proof.